micro-rdk/src/common/ota.rs

```diff
@@ -103,20 +103,37 @@ struct EspAppDesc {
     reserv2: [u32; 20],
 }
 
+#[derive(Error, Debug)]
+pub(crate) enum ConfigError {
+    #[error("version {0} has length {1}, maximum allowed characters is {2}")]
+    InvalidVersionLen(String, usize, usize),
+    #[error("the given url {0}, is not a valid URI: {1}")]
+    InvalidUrl(String, String),
+    #[error(transparent)]
+    AttributeError(#[from] AttributeError),
+    #[error("config missing attribute `{0}`")]
+    MissingAttribute(String),
+    #[error("config missing value for `{0}`")]
+    MissingValue(String),
+    #[error("{0}")]
+    Other(String),
+}
+
 // TODO(RSDK-9214): surface (and use) underlying errors properly, use transparent where possible
 #[derive(Error, Debug)]
-pub enum OtaError {
+pub(crate) enum OtaError {
     #[error("{0}")]
-    ConfigError(String),
-    #[cfg(feature = "esp32")]
+    ConfigError(#[from] ConfigError),
     #[error(transparent)]
-    EspError(#[from] EspError),
+    NetworkError(#[from] hyper::Error),
+    #[error("error getting the writer handle for OTA: {0}")]
+    HandleError(String),
```

Comment:
I don't love `HandleError` - the error isn't really so much about handles, as it is the ability to use the handle to update. So, maybe `WriteError` or `UpdateError`?